loki/transform/transform_hoist_variables.py

```diff
@@ -240,7 +243,7 @@ def transform_subroutine(self, routine, **kwargs):
             routine.arguments += hoisted_temporaries
 
         call_map = {}
-        for call in FindNodes(CallStatement).visit(routine.body):
+        for call in FindNodes(CallStatement).visit(routine.body) + list(FindInlineCalls().visit(routine.body)):
```

Comment:
We have a bit of a mismatch there, something we should clean up separately:
- `FindNodes` has indeed always returned lists. Since these are built incrementally and can be quite long, there could indeed a performance impact from using tuples
- Expression visitors return by default sets, unless `unique=False` is used.